### PR TITLE
Convert actor skills and certain roll options to longform

### PR DESCRIPTION
**Pull request by CarlosFdez**
We also add some shims to map the existing short form names to the new long form stuff. Once we're fully done migrating everything we can make them show deprecation warnings with a very fast expiration (6.2 or 6.3) just in case we miss anything..

From a quick test, Raging Intimidation, Rogue Dedication, Acrobat Dedication, and Skill Mastery all continued to function, even with some of them in an incomplete migration state.

### CODE REVIEW

File: src/module/actor/creature/data.ts [hunk 1]
```diff
@@ -91,7 +91,7 @@ interface CreatureSystemData extends Omit<CreatureSystemSource, "attributes">, A
     /** Saving throw data */
     saves: CreatureSaves;
 
-    skills: Record<string, SkillData>;
+    skills: Record<SkillLongForm, SkillData>;
```
Comment (stwlam) [inline src/module/actor/creature/data.ts:94]: Where did lore go?

Comment (CarlosFdez) [reply src/module/actor/creature/data.ts:94]: Character already doesnt account for lore in the type. That said, i could see about improving the type.

Comment (stwlam) [reply src/module/actor/creature/data.ts:94]: I think the only other option is `string`, unfortunately.

Comment (CarlosFdez) [reply src/module/actor/creature/data.ts:94]: Record<SkillLongForm, SkillData> & Record<string, SkillData | undefined> wouldn't work?

File: src/module/actor/character/document.ts [hunk 2]
```diff
@@ -742,9 +742,10 @@ class CharacterPF2e<TParent extends TokenDocumentPF2e | null = TokenDocumentPF2e
             rollOptionsAll[`attribute:${key}:mod:${mod}`] = true;
         }
 
-        for (const key of SKILL_ABBREVIATIONS) {
+        for (const key of SKILL_LONG_FORMS) {
```
Comment (stwlam) [inline src/module/actor/character/document.ts:745]: Can we rename to `SKILL_SLUGS` now? :grin:

File: src/module/actor/character/document.ts [hunk 3]
```diff
@@ -742,9 +742,10 @@ class CharacterPF2e<TParent extends TokenDocumentPF2e | null = TokenDocumentPF2e
...
             const rank = this.system.skills[key].rank;
             rollOptionsAll[`skill:${key}:rank:${rank}`] = true;
+            rollOptionsAll[`skill:${SKILL_DICTIONARY_REVERSE[key]}:rank:${rank}`] = true;
```
Comment (stwlam) [inline src/module/actor/character/document.ts:748]: Mind commenting that this is for backward compatibility?

Comment (CarlosFdez) [reply src/module/actor/character/document.ts:748]: Sure.

File: src/module/actor/character/document.ts (hunk 2 above)
Comment (CarlosFdez) [reply src/module/actor/character/document.ts:745]: Not yet but that would be great to do after the next step. I want to keep the diffs as small as I can for now to not overwhelm the review process.

**Review by stwlam — DISMISSED** [review]

**Review by stwlam — APPROVED** [review]

**Pull request merged**